File: src/types/process.ts
/**
 * 文件名称：process.ts
 *
 * 文件功能：
 * 定义进程详情页数据结构。
 *
 * 主要职责：
 * - 约束进程元数据与资源占用字段
 * - 约束释放端口的业务结果
 *
 * 作者：Chushi Jack
 * 创建时间：2026-08-20
 */

export interface ProcessDetail {
  pid: number;
  name: string;
  displayName: string;
  serviceLabels: string[];
  status: string;
  user: string;
  startTime: string;
  duration: string;
  priority: string;
  path: string;
  command: string;
  cpuUsage: number;
  memoryBytes: number;
  ports: number[];
}

export type KillMode = "graceful" | "force";

export type ControllerKind =
  | "windowsService"
  | "launchDaemon"
  | "launchAgent"
  | "application"
  | "supervisor"
  | "dockerContainer"
  | "unknown";

/** 后端释放端口的结果：成功、被控制者重新拉起、或需到控制者处停止。 */
export type KillOutcome =
  | {
      kind: "released";
      process: string;
      controller: string;
      controllerKind: string;
    }
  | {
      kind: "processRestarted";
      process: string;
      controller: string;
      controllerKind: string;
      pid: number;
    }
  | {
      kind: "requiresExternalStop";
      process: string;
      controller: string;
      controllerKind: string;
    }
  | {
      kind: "requireControllerStop";
      process: string;
      controller: string;
      controllerKind: string;
    };

File: src-tauri/src/models/process.rs
//! 文件名称：process.rs
//!
//! 文件功能：
//! 定义进程详情页所需的数据结构。
//!
//! 主要职责：
//! - 汇总进程元数据、资源占用和关联端口
//!
//! 作者：Chushi Jack
//! 创建时间：2026-08-20

use serde::{Deserialize, Serialize};

/// 进程详情，对应详情页卡片与性能条。
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProcessDetail {
    pub pid: u32,
    pub name: String,
    pub display_name: String,
    /// 按占用端口识别的开发服务名，例如 Vite Dev Server。
    pub service_labels: Vec<String>,
    pub status: String,
    pub user: String,
    pub start_time: String,
    pub duration: String,
    pub priority: String,
    pub path: String,
    pub command: String,
    pub cpu_usage: f32,
    pub memory_bytes: u64,
    pub ports: Vec<u16>,
}
